modules/nlu/src/backend/api.ts

```diff
@@ -54,12 +54,25 @@ export default async (bp: typeof sdk, nlus: EngineByBot) => {
     }
   }
 
+  router.get('/currentModelHash', async (req, res) => {
+    const engine = nlus[req.params.botId] as ScopedEngine
+    if (engine.modelHash) {
+      res.send(engine.modelHash)
+      return
+    }
+    const intents = await engine.storage.getIntents()
+    const modelHash = await engine.computeModelHash(intents)
+    res.send(modelHash)
+  })
+
   router.get('/confusion/:modelHash', async (req, res) => {
+    const engine = nlus[req.params.botId] as ConfusionEngine
+    const confusionComputing = engine.confusionComputing
     try {
-      const matrix = await (nlus[req.params.botId] as ScopedEngine).storage.getConfusionMatrix(req.params.modelHash)
-      res.send(matrix)
+      const matrix = await engine.storage.getConfusionMatrix(req.params.modelHash)
+      res.send({ matrix, confusionComputing })
     } catch (err) {
-      res.sendStatus(401)
+      res.send({ confusionComputing })
```

Comment:
Shouldn't something be logged if there's an error? If errors can be ignored, please add a comment stating why